### PR TITLE
Feature/admin user role crud

### PR DESCRIPTION
**Pull request by zakhenry**

### CODE REVIEW

File: app/src/app/admin/users/editUser/roles/roles.spec.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                    roles: roles,
+                });
+            });
+
+        });
+
+        describe('User Roles admin', () => {
+
+
+            it('should do something', () => {
```
Comment (ghost) [inline app/src/app/admin/users/editUser/roles/roles.spec.ts]: Is this meant to be written?

Comment (zakhenry) [inline app/src/app/admin/users/editUser/roles/roles.spec.ts]: Yea probably! will do.

File: app/src/common/services/role/roleService.spec.ts [hunk 2]
```diff
@@ -0,0 +1,342 @@
+namespace common.services.user {
+
+    describe('UserService', () => {
```
Comment (ghost) [inline app/src/common/services/role/roleService.spec.ts]: Name space and name of tests need to be updated? 'common.services.role' and 'Role Service'?

File: app/src/common/services/role/roleService.spec.ts [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+            });
+
+        });
+
+        afterEach(() => {
+            $httpBackend.verifyNoOutstandingExpectation();
+            $httpBackend.verifyNoOutstandingRequest();
+        });
+
+        describe('Initialisation', () => {
```
Comment (ghost) [inline app/src/common/services/role/roleService.spec.ts:27]: Are these tests just copied for user service? Haven't been updated to tests for role service yet?

File: app/src/common/services/user/userService.ts [hunk 4]
```diff
@@ -274,6 +274,32 @@ namespace common.services.user {
...
+                return _.pick(role, 'key');
+            });
+
+            let flattenedRoles = _.pluck(roleData, 'key');
+
+            if (_.xor(flattenedRoles, user.roles).length === 0){
+                return this.$q.when(user);
+            }
+
+            console.log('user.roles, flattened roles', user.roles, flattenedRoles);
```
Comment (ghost) [inline app/src/common/services/user/userService.ts]: Debug?

File: app/src/common/services/role/roleService.spec.ts (hunk 3 above)
Comment (zakhenry) [inline app/src/common/services/role/roleService.spec.ts:27]: Yup fixed

**Pull request merged**